Types/Module.cs

```diff
@@ -403,7 +403,20 @@ public static CkanModule FromJson(string json)
             }
             // All good! Return module
             return newModule;
-        }            
+        }
+        public override bool Equals(object obj)
+        {
+            var other = obj as Module;
+            return other != null
+            ? identifier.Equals(other.identifier) && version.Equals(other.version)
+            : base.Equals(obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return identifier.GetHashCode();
```

Comment:
I'm not fluent with `GetHashCode` mechanics, but wouldn't this mean that two modules hash to the same bucket if they have the same identifier? As expressed in the `Equals` code above, we really consider them to be the same if they have both the same identifier *and* the same version.